DeepSpeech.py

```diff
@@ -608,6 +615,7 @@ def __call__(self, progress, data, **kwargs):
                     else:
                         raise
                 except tf.errors.OutOfRangeError:
+                    exception_box.raise_if_set()
```

Comment:
Why are we swallowing all exceptions in the dataset generator and only re-raising in this specific case here?

Reply:
This took me quite a while to "fix" and maybe you got some better approach: Without this code all kinds of exceptions raised in the generator (which can now happen - your original code prepared all critical things before the generator) will just interrupt generation and reult int this `OutOfRangeError`. I was not able to get any meaningful debug information from generator-failures (try it by raising something inside it). This "second chance" raising helped a lot. I am open for better solutions...

Reply:
OK, so the goal is to surface meaningful error information from the generator to the user. And you're saying no matter what type of exception is raised in the generator, it always gets turned into an OutOfRangeError here.

Maybe move this `raise_if_set` call outside of the except so an exception will always be raised if it exists. Otherwise we risk silently ignoring some edge case error, or if TF behavior changes after an upgrade.

Reply:
Added another one directly following the `session.run` call, as the current one has to stay where it is, as it has to be called before the `break`.